### PR TITLE
BUG: sets in str.cat

### LINKED ISSUE #23009: sets in str.cat?!
In #20347, I was asked to allow list-like inputs for the `.str.cat` methods for Series. However, I didn't know that `is_listlike({'a', 'b', 'c'})` is True (for sets) until I stumbled over it in #22486.

These objects should imo clearly be disallowed, lest users shoot themselves in the foot massively:
```
>>> s = pd.Series(['a', 'b', 'c'])
>>> s.str.cat({'a', 'b', 'c'})
0    ac
1    bb
2    ca
dtype: object
```

I'm not even sure this should go through a deprecation cycle - IMO it's just a plain ol' bug that should be fixed ASAP.

### PR DESCRIPTION
**Pull request by h-vetinari**
- [x] closes #23009
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

This is split off from #23065.

### CODE REVIEW

**Comment by h-vetinari** [comment]
@jreback 
Tiny follow-up to / split-off from #23065, all green and ready for review

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -879,7 +879,7 @@ Numeric
 Strings
 ^^^^^^^
 
--
+- Bug in :meth:`Series.str.cat` which falsely did not raise if `others` was a `set` (:issue:`23009`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: this is an api breaking change, so move it there. make this declerative as well, e.g. Series.str.cat will now raise if others is a set

File: pandas/tests/test_strings.py [hunk 2]
```diff
@@ -301,7 +301,15 @@ def test_str_cat_mixed_inputs(self, box):
         with tm.assert_raises_regex(TypeError, rgx):
             s.str.cat([u, [u, d]])
 
-        # forbidden input type, e.g. int
+        # forbidden input type: set
```
Comment (jreback) [inline pandas/tests/test_strings.py]: add the issue number here

**Review by jreback — CHANGES_REQUESTED** [review]
minor comments, ping on green.

**Comment by jreback** [comment]
thanks!

**Pull request merged**